### PR TITLE
Unit test fixes -- run as normal user and GH action

### PR DESCRIPTION
**Pull request by vojtechtrefny**
This should be a final fix for #870 In addition to the changes to the unit tests, I have added a GH action to run the unit tests on Ubuntu so we can be sure we won't break this in the future.

### CODE REVIEW

File: blivet/util.py [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ def detect_virt():
     except (safe_dbus.DBusCallError, safe_dbus.DBusPropertyError):
         return False
     else:
-        return vm[0] in ('qemu', 'kvm', 'xen')
+        return vm[0] in ('qemu', 'kvm', 'xen', 'microsoft')
```
Comment (VladimirSlavik) [inline blivet/util.py:1163]: What does this actually do? Enable running tests under WSL? Could be useful to know, perhaps in the commit message :)

**Review by VladimirSlavik — COMMENTED** [review]
I won't ack since I don't know this too well, but code LGTM.

Did you consider running the unit tests under Fedora too? It would not be as straightforward as the action you have here, but to me it sounds like a far more important target to keep hitting.

FWIW, there's a [GHA key](https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container) to run things in a container on top of the runs-on, so it might be a one-liner-ish thing. Unfortunately I never tried making such containerized run into a matrix so it would become a "for free" thing...

**Comment by vojtechtrefny** [comment]
> Did you consider running the unit tests under Fedora too? It would not be as straightforward as the action you have here, but to me it sounds like a far more important target to keep hitting.

We already run the unit tests in our CI (and we might use Packit for Fedora in the future), but we run the entire test suite as root (basically just running `sudo make test`). This action is here to make sure that unit tests can run without root privileges so running it only on Ubuntu should be fine (especially because that's where we have the most problems, because of the `PATH` differences in Debian based distributions).

File: tests/unit_tests/devices_test/lvm_test.py [hunk 2]
```diff
@@ -651,41 +652,49 @@ def greeting(self, val):  # pylint: disable=arguments-differ
         self.assertEqual(c.greeting, "Set by A: Welcome")
 
 
+class BlivetLVMUnitTest(unittest.TestCase):
+
+    @patch("blivet.formats.fs.Ext4FS.supported", return_value=True)
+    @patch("blivet.formats.fs.Ext4FS.formattable", return_value=True)
+    def setUp(self, *args):  # pylint: disable=unused-argument
+        self.b = blivet.Blivet()
```
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply tests/unit_tests/devices_test/lvm_test.py]: Ah, that's setUpClass - sorry!

**Pull request merged**